android/src/main/java/com/pylon/RNSpokestack/RNSpokestackModule.java

```diff
@@ -128,13 +146,46 @@ public void synthesize (ReadableMap ttsInput) {
         }
     }
 
+    @ReactMethod
+    public void classify (String utterance, ReadableMap context) {
+        AsyncResult<NLUResult> asyncResult = nlu.classify(utterance, new NLUContext(this.pipeline.getConfig()));
+        asyncResult.registerCallback(this);
+    }
+
+    @Override
+    public void call(NLUResult arg) {
+        WritableMap react_event = Arguments.createMap();
+        WritableMap result = Arguments.createMap();
+        WritableMap slots = Arguments.createMap();
+        for (Map.Entry<String, Slot> entry : arg.getSlots().entrySet()) {
+            WritableMap slot = Arguments.createMap();
+            Slot s = entry.getValue();
+            slot.putString("type", s.getName());
+            slot.putString("value", s.getValue().toString());
```

Comment:
:| Javascript, undoing all the hard work of static types.

Reply:
it helps to think of this a serialization rather than losing typing…

Reply:
Fair enough. Slightly less sad now.